### PR TITLE
Add "Modding APIs" section

### PR DESCRIPTION
**Pull request by legoandmars**
The wiki needs a definitive source of user-created APIs for common usecases, as discussed in #28 

This PR adds a modding API section with:
- A section for non-code APIs so non-programmers can have a section to check for simple swaps (like audio replacement, custom suits, etc)
- A section for code APIs so modders can have an easy overview

I think going forward, at least for now, the sections should be:
- Initial setup (Max's guide on how to get everything initialized)
- Modding APIs (for user-created APIs, such as customsounds)
- Advanced Modding (for specific topics like Networking and creating Configs. the section may need a rename)

Here's an example of what I think the Modding APIs section could look like once we start getting some guides (like #28)
image

### CODE REVIEW

**Comment by legoandmars** [comment]
would love any comments on this @MaxWasUnavailable @Xilophor since I'm not 100% sure the best way to go about this

I want to keep the playing field as fair as possible, but I do think we should incentivize modders to write guides on their APIs - I've done this in the PR by adding a gold star to any on-wiki modding APIs

**Comment by legoandmars** [comment]
there might also be some APIs missing right now, feel free to suggest/add some. I just did a quick general overview of APIs I've seen mentioned a lot

**Comment by MaxWasUnavailable** [comment]
I feel like advanced topics should come before APIs. Since we then have a fairly clear logical ordering going from non-API to API stuff, rather than no/yes/no.

**Comment by Xilophor** [comment]
I was thinging about this yesterday. With all of this being added, the main non-developer stuff is being "squished" in the sidebar; should we look into separating it into a Beginner's Guide and a Developer's Guide - may be appropriate for another PR/issue

**Comment by legoandmars** [comment]
> I was thinging about this yesterday. With all of this being added, the main non-developer stuff is being "squished" in the sidebar; should we look into separating it into a Beginner's Guide and a Developer's Guide - may be appropriate for another PR/issue

I think we can have two separate sidebars like (bsmg.wiki)[https://github.com/bsmg/wiki] does, I can look into it in a later PR

**Comment by MaxWasUnavailable** [comment]
> > I was thinging about this yesterday. With all of this being added, the main non-developer stuff is being "squished" in the sidebar; should we look into separating it into a Beginner's Guide and a Developer's Guide - may be appropriate for another PR/issue
> 
> I think we can also have two separate sidebars like [bsmg.wiki](https://github.com/bsmg/wiki) does, I can look into it in a later PR 

This would be good imo. Separate mod creation and mod usage.

**Comment by MaxWasUnavailable** [comment]
> I'm also honestly not 100% sure what the line between an API and advanced modding topic is, that might have to be figured out as we go
> 
> Like, the page on custom networking uses UnityNetcodeWeaver, which I would *technically* class as an API, but the concept is so fundamental to modding 
> 

We can work this out as we go. We may want to eventually boil it down to the fundamentals, and link to APIs that make it easier.

**Comment by legoandmars** [comment]
> This would be good imo. Separate mod creation and mod usage.

Yeah, I think the sidebar is probably *fine* right this moment, but once #44 #38 and #28 are merged it's going to start getting overwhelming if we keep adding more pages

for now I've made an issue - #47

File: docs/apis/modding-apis.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## Adding APIs to the wiki
+
+If you're an API developer and want to get your API added to this list, feel free to make a [pull request](https://github.com/LethalCompany/ModdingWiki), assuming your API meets the following criteria:
+
+- Open source
+- Fairly well documented - even better if you write a wiki article*
+<!-- I would love a way to do a superscript citation here but I don't know if you can by default? -->
+\* Some exceptions may be made if it's an especially groundbreaking API, or it's the only existing solution.
```
Comment (Xilophor) [inline docs/apis/modding-apis.md:64]: <sup></sup> is a html tag that can do superscript

Comment (legoandmars) [reply docs/apis/modding-apis.md:64]: oh right i forgot you can use HTML temporarily

**Review by Xilophor — DISMISSED** [review]
I think we've already discussed everything relating to this PR, so I'll go ahead and approve.

**Pull request merged**